build-config/Makefile

```diff
@@ -162,7 +162,7 @@ ifneq (,$(MAKECMDGOALS))
       $(error Unable to find valid machine configuration directory.)
     endif
     include $(MACHINEDIR)/machine.make
-    ARCHDIR ?= $(realpath ./arch)
+    ARCHDIR ?= $(realpath ./arch/$(ONIE_ARCH))
```

Comment:
Thanks for reviewing my changes
1. i have compiled qemu_armv7a and arm64 toolchain aarch64 successfully with these changes.
2. i have corrected the mistake with x86_64 architecture.